test/Microsoft.Azure.Devices.Edge.Test/SetupFixture.cs

```diff
@@ -95,39 +96,24 @@ public async Task Setup()
         }
 
         [OneTimeTearDown]
-        public async Task Teardown()
-        {
-            try
-            {
-                await Profiler.Run(
-                    async () =>
+        public Task AfterAllAsync() => TryFinally.DoAsync(
+            () => Profiler.Run(
+                async () =>
+                {
+                    using (var cts = new CancellationTokenSource(Context.Current.TeardownTimeout))
                     {
-                        using (var cts = new CancellationTokenSource(Context.Current.TeardownTimeout))
-                        {
-                            CancellationToken token = cts.Token;
-
-                            await this.daemon.StopAsync(token);
+                        CancellationToken token = cts.Token;
 
-                            Assert.IsNotNull(this.device);
-                            await this.device.MaybeDeleteIdentityAsync(token);
-                        }
-                    },
-                    "Completed end-to-end test teardown");
-            }
+                        await this.daemon.StopAsync(token);
```

Comment:
If there is an exception during OneTimeSetup, iotedge service is not installed; I belive OneTimeTearDown will still trigger, right? then will StopAsync in OneTimeTearDown throws exception (at least ServiceController on Windows implementation may throw)? if so, what will be shown in the log?

Reply:
From https://github.com/nunit/docs/wiki/OneTimeTearDown-Attribute:

> So long as any OneTimeSetUp method runs without error, the OneTimeTearDown method is guaranteed to run. It will not run if a OneTimeSetUp method fails or throws an exception.